File: day2/day2_1.go
package Day2

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

func isSafe(values []int) bool {
	incr, decr := false, false

	for i := 1; i < len(values); i++ {
		left := values[i-1]
		right := values[i]

		if left < right {
			incr = true
		} else if left > right {
			decr = true
		}

		if incr && decr {
			return false
		}

		if abs(left-right) < 1 || abs(left-right) > 3 {
			return false
		}
	}
	return true
}

func Day2_1() int {
	file, _ := os.Open("day2/input.txt")
	defer file.Close()

	scanner := bufio.NewScanner(file)
	count := 0

	for scanner.Scan() {
		line := scanner.Text()
		parts := strings.Fields(line)

		values := make([]int, len(parts))
		for i, part := range parts {
			values[i], _ = strconv.Atoi(part)
		}

		if isSafe(values) {
			count++
		}
	}

	return count
}

File: day2/day2_1.py
def day2_1():
    count = 0
    with open("day2/input.txt", "r") as file:
        for line in file:
            values = line.split()
            flag = True
            incr, decr = False, False

            for i in range(1, len(values)):
                left = int(values[i - 1])
                right = int(values[i])

                if left < right:
                    incr = True
                elif left > right:
                    decr = True

                if incr and decr:
                    flag = False
                    break

                if abs(left - right) < 1 or abs(left - right) > 3:
                    flag = False
                    break

            if flag:
                count += 1
        print("day2.1", count)
